### PR TITLE
enh(scala) fix triple quoted string

### LINKED ISSUE #2986: (Scala) Strings in triple quotes are highlighted incorrectly.
**Describe the issue**
For example, the code:

```
val s = """ this is a string "this is still a string" another quote: " after the quote """
```

create a String with the content:

```
 this is a string "this is still a string" another quote: " after the quote 
```
This code is highlighted incottectly. `this is still a string`, and `after the quote` is not formatted as a string, although they should be. The rule is that single quotes are just part of the string when they are surrounded by triple quotes.

For more context please see [this](https://meta.stackoverflow.com/q/404946/2359227) Meta Stack Overflow question.

**Which language seems to have the issue?**
Scala

**Are you using `highlight` or `highlightAuto`?**

I am not sure. The one that Stack Overflow is using.

**Sample Code to Reproduce**
```
val s = """ this is a string "this is still a string" another quote: " after the quote """
```

**Expected behavior**
`s` to be string, and not to open code segments in the middle of the string.

**Additional context**
Nothing I can think of :)

### PR DESCRIPTION
**Pull request by joshgoebel**
Resolves #2986

### Checklist
- [x] Added markup tests, or they don't apply here because...
- [x] Updated the changelog at `CHANGES.md`
- [x] Added myself to `AUTHORS.txt`, under Contributors

### CODE REVIEW

File: src/languages/scala.js [hunk 1]
```diff
@@ -29,23 +29,24 @@ export default function(hljs) {
...
       {
         begin: '"',
         end: '"',
         illegal: '\\n',
         contains: [ hljs.BACKSLASH_ESCAPE ]
       },
-      {
-        begin: '"""',
-        end: '"""',
-        relevance: 10
```
Comment (toshetah) [inline src/languages/scala.js:41]: Basically the main change I see here is that you removed the `relevance`, and that you changed the order. Can you please explain what do those changes do? Next time I might just contribute it myself :)

**Review by toshetah — APPROVED** [review]

File: src/languages/scala.js (hunk 1 above)
Comment (joshgoebel) [reply src/languages/scala.js:41]: Well if `"` matches first that breaks everything, so we have to match `"""` first... and the relevance is a much longer explication... find the thread on huge relevance and read it for a start for more thoughts... 10 is just too high for something that is shared by several languages now, so I removed the huge relevance bump.

_The sequencing is what fixes the issue._

**Review by toshetah — APPROVED** [review]

**Comment by toshetah** [comment]
I'd add another test of multi line string:

```
val multiLineString =
  """aaa
    |ddd
    |sdfa""".stripMargin
```

should produce:

```
aaa
ddd
sdfa
```

and:

```
```
val multiLineString =
  """aaa
    |ddd
    |sdfa"""
```

should produce:

```
aaa
    |ddd
    |sdfa
```

Code run at [Scastie](https://scastie.scala-lang.org/N2V8tPHcQguXehJyhxD4pA).

**Comment by joshgoebel** [comment]
If you wanted to contribute additional tests to the project (expanding on `strings.txt`) that'd be great... long-term it'd be nice to test all the different types of strings, escaping of special chars, etc...

**Comment by joshgoebel** [comment]
> I'd add another test of multi line string:

Of course our parsing engine doesn't care about the how the string is interpreted, we just highlight it. So your last two examples are irrelevant in our context - as far as I can tell. The highlighted is the same in both cases, regardless of what the end result it inside the runtime.

**Pull request merged**